src/FSharp.AWS.DynamoDB/TableContext.fs

```diff
@@ -1402,6 +1318,141 @@ type TableContext<'TRecord>
         else
             t.VerifyTableAsync()
 
+    member t.Transaction() =
+        match metricsCollector with
+        | Some metricsCollector -> Transaction(metricsCollector = metricsCollector)
+        | None -> Transaction()
+
+/// <summary>
+///    Represents a transactional set of operations to be applied atomically to a arbitrary number of DynamoDB tables.
+/// </summary>
+/// <param name="metricsCollector">Function to receive request metrics.</param>
+and Transaction(?metricsCollector: (RequestMetrics -> unit)) =
+    let transactionItems = ResizeArray<TransactWriteItem>()
+    let mutable (dynamoDbClient: IAmazonDynamoDB) = null
+
+    let setClient client =
+        if dynamoDbClient = null then
```

Comment:
I think this is bad news - better to be explicit that a transaction is going to need a client
Force people to either pick a relevant one,
or expose it on the Table so they can call get it from there
or have a `table.TransactWriteItems(transaction)` that makes it clear where it's coming from

Reply:
We went backwards and forwards on this a lot. I have struggled to get a good grasp of when people would use multiple clients (and how we would be able to check multiple clients are being used properly). If we need to make this explicit I would have preferred to go down the `MultiTableContext` path, but it seems like a lot of difficult-to-discover ceremony for a (presumably) vanishingly small number of API consumers.

Perhaps the best middle ground is `new Transaction(client, processMetrics)` as well as `table.NewTransaction()`? The latter would work fine for the bulk of consumers that only have one client & don’t want to pass it around separately to the tables.

Reply:
Yes, I think both mechanisms is fine, as the 90% case is to do table.Transact/NewTransaction

Its more about Explicit Is Better than Explicit wrt the mechanism of how the client is derived when you create it independently

Don't like `New`, and I think FDG outlaws it, but I guess `Transact` is a bit much if it's not actually going to `Transact` directly - maybe `CreateTransaction`?
I was thinking about a `Transaction.Attach(table)` or a `Transaction(table)`, but ultimately that coupling/semantic is confusing too.

Similarly BeginTransaction would be confusing as nothing is really beginning (and it's not beginning from the Table end

I guess this is why I punted on it all in the first instance.

For my own use, as long as there's a Tranaction ctor with a mandatory client arg and an optional ctor, I'll pass that around as a `Func<Transaction>` and wire it up in the same way that I have a `Func<TableContext>` now

I guess if there was a ClientContext that held the client and the collector, then you could pass that common thing to the TableContext and TransactionContext ctors, but that's a bigger job and it's a big jump from a library that had one abstraction (TableContext) to suddenly having 3. (Though the chances are it does make sense overall - in Equinox I have DynamoStoreClient that wraps the `IAmazonDynamoDb`, but its definitely at the other end of the separating things out vs whacking it all into one type spectrum)

Reply:
`CreateTransaction` sounds good to me